mmstudio/src/main/java/org/micromanager/data/internal/multipagetiff/MultipageTiffReader.java

```diff
@@ -504,16 +582,22 @@ private Image readImage(IFDData data) throws IOException {
          // a nio buffer directly as the Image storage (even better if memory
          // mapped).
          switch (pixelType) {
-            case GRAY8:
-               return new DefaultImage(pixelBuffer.array(), formatPmap,
+            case GRAY8: {
+               Image img = new DefaultImage(pixelBuffer.array(), formatPmap,
                      coords, metadata);
-            case GRAY16:
+               tryRecycleLargeBuffer(pixelBuffer);
+               return img;
```

Comment:
These can be factored out after the `switch` statement to avoid repetition (for each `case`).